src/nupic/regions/SDRClassifierRegion.py

```diff
@@ -0,0 +1,500 @@
+          dataType='UInt32',
+          required=True,
+          count=1,
+          constraints='',
+          # arbitrarily large value for backward compatibility 
+          defaultValue=1000,
+          accessMode='Create'),
+
+        steps=dict(
+          description='Comma separated list of the desired steps of '
+                      'prediction that the classifier should learn',
+          dataType="Byte",
+          count=0,
+          constraints='',
+          defaultValue='0',
+          accessMode='Create'),
+
+        alpha=dict(
+          description='The alpha used to compute running averages of the '
+                      'bucket duty cycles for each activation pattern bit. A '
+                      'lower alpha results in longer term memory',
+          dataType="Real32",
+          count=1,
+          constraints='',
+          defaultValue=0.001,
+          accessMode='Create'),
+
+        implementation=dict(
+          description='The classifier implementation to use.',
+          accessMode='ReadWrite',
+          dataType='Byte',
+          count=0,
+          constraints='enum: py, cpp'),
+
+        clVerbosity=dict(
+          description='An integer that controls the verbosity level, '
+                      '0 means no verbose output, increasing integers '
+                      'provide more verbosity.',
+          dataType='UInt32',
+          count=1,
+          constraints='',
+          defaultValue=0,
+          accessMode='ReadWrite'),
+
+      ),
+      commands=dict()
+    )
+
+    return ns
+
+
+  def __init__(self,
+               steps='1',
+               alpha=0.001,
+               clVerbosity=0,
+               implementation=None,
+               maxCategoryCount=None
+               ):
+
+    # Set default implementation
+    if implementation is None:
+      implementation = Configuration.get('nupic.opf.sdrClassifier.implementation')
+
+    # Convert the steps designation to a list
+    self.classifierImp = implementation
+    self.steps = steps
+    self.stepsList = eval("[%s]" % (steps))
+    self.alpha = alpha
+    self.verbosity = clVerbosity
+
+    # Initialize internal structures
+    self._sdrClassifier = SDRClassifierFactory.create(
+      steps=self.stepsList,
+      alpha=self.alpha,
+      verbosity=self.verbosity,
+      implementation=implementation,
+    )
+    self.learningMode = True
+    self.inferenceMode = False
+    self.maxCategoryCount = maxCategoryCount
+    self.recordNum = 0
+    self._initEphemerals()
+
+    # Flag to know if the compute() function is ever called. This is to 
+    # prevent backward compatibilities issues with the customCompute() method
+    # being called at the same time as the compute() method. Only compute() 
+    # should be called via network.run(). This flag will be removed once we 
+    # get to cleaning up the clamodel.py file.
+    self._computeFlag = False
+
+
+  def _initEphemerals(self):
```

Comment:
I added a doc string here. This seems to be a common function that exists in all Regions. The function is supposed to "Initialize ephemeral attributes that are not saved with the checkpoint. These ephemeral attributes may be used by derived classes." 

I don't think there are any ephemeral attributes in SDRClassifierRegion, and implement it as `pass`